dask/dataframe/io/parquet.py

```diff
@@ -397,22 +394,23 @@ def _write_pyarrow(df, path, write_index=None, append=False,
     template = fs.sep.join([path, 'part.%i.parquet'])
 
     write = delayed(_write_partition_pyarrow)
-    first_kwargs = kwargs.copy()
-    first_kwargs['metadata_path'] = fs.sep.join([path, '_metadata'])
-    writes = [write(part, open_with, template % i, write_index,
-                    **(kwargs if i else first_kwargs))
+    metadata_path = fs.sep.join([path, '_metadata'])
+    writes = [write(part, open_with, template % i, write_index, compression, metadata_path,
+                    **kwargs)
               for i, part in enumerate(df.to_delayed())]
     return delayed(writes)
 
 
-def _write_partition_pyarrow(df, open_with, filename, write_index,
+def _write_partition_pyarrow(df, open_with, filename, write_index, compression=None,
```

Comment:
I'd handle this fix by reverting this change (leave `compression` in `kwargs`, and do:

`kwargs.pop('compression')` before passing to `write_metadata`. This ensures that compression only gets passed to the backend if it's specified (so that `'default'` still works, and removes it from being passed to `write_metadata`.

Reply:
done

Reply:
done